fix(@wdio/appium-service): add port to appium start command

vparlah opened the pull request:
## Proposed changes
Fix for this PR: **(@wdio/appium-service): make Appium start on a random port** https://github.com/webdriverio/webdriverio/pull/11757. The port value wasn't added in the appium start command. So, no matter what random port was chosen, appium always tries to start it on default, even if it's busy

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Polish (an improvement for an existing feature)
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (improvements to the project's docs)
- [ ] Specification changes (updates to WebDriver command specifications)
- [ ] Internal updates (everything related to internal scripts, governance documentation and CI files)

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Backport Request

[//]: # (The current `main` branch is the development branch for WebdriverIO v9. If your change should be released to the current major version of WebdriverIO (v8), please raise another PR with the same changes against the `v8` branch.)

- [ ] This change is solely for `v9` and doesn't need to be back-ported
- [x] Back-ported PR at https://github.com/webdriverio/webdriverio/pull/13179

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### Reviewers: @webdriverio/project-committers

christian-bromann commented:
@vparlah mind signing the CLA?

christian-bromann commented:
@vparlah any updates?

vparlah commented:
@christian-bromann 
1) Signed! 
2) I fixed existing and added some new tests
3) I was able to build and run the v8 project locally(but with npm only). However, I had issues with running v9... so I just copied tests I did for v8

christian-bromann commented:
> 3\. However, I had issues with running v9

Mind sharing these issues? Note that switching between them may be not working as we transitioned to pnpm so you would have to clear out all `node_modules` etc.

vparlah commented:
I rebased this branch on the latest main, cleaned everything, and it started to work fine.
I fixed windows related unit tests. When you get a chance trigger the build again for this and v8(https://github.com/webdriverio/webdriverio/pull/13179 ), please.

vparlah commented:
There are some component test failures... Do you think they are related to this change?

christian-bromann commented:
> Do you think they are related to this change?

No they aren't, I will go ahead and merge

christian-bromann reviewed (approved):
LGTM 👍
